.github/workflows/continuous-integration.yml

```diff
@@ -127,9 +132,37 @@ jobs:
         & watchman --version
         echo "PATH=$ENV:PATH" >>$env:GITHUB_ENV
 
-    - name: Functional test
+    - id: functional_test
+      name: Functional test
       shell: bash
       run: |
+        export GIT_TRACE2_EVENT="$PWD/$TRACE2_BASENAME/Event"
+        export GIT_TRACE2_PERF="$PWD/$TRACE2_BASENAME/Perf"
+        export GIT_TRACE2_EVENT_BRIEF=true
+        export GIT_TRACE2_PERF_BRIEF=true
+        mkdir -p "$TRACE2_BASENAME"
+        mkdir -p "$TRACE2_BASENAME/Event"
+        mkdir -p "$TRACE2_BASENAME/Perf"
+        git version --build-options
         cd ../out
         PATH="$PWD/Scalar/$BUILD_FRAGMENT:$PWD/Scalar.Service/$BUILD_FRAGMENT:$PATH"
         Scalar.FunctionalTests/$BUILD_FRAGMENT/Scalar.FunctionalTests$BUILD_FILE_EXT --test-scalar-on-path --full-suite
+
+    - id: trace2_zip_unix
+      if: runner.os != 'Windows' && ( success() || failure() ) && ( steps.functional_test.conclusion == 'success' || steps.functional_test.conclusion == 'failure' )
+      name: Zip Trace2 Logs (Unix)
+      shell: bash
+      run: zip -q -r $TRACE2_BASENAME.zip $TRACE2_BASENAME/
+
+    - id: trace2_zip_windows
+      if: runner.os == 'Windows' && ( success() || failure() ) && ( steps.functional_test.conclusion == 'success' || steps.functional_test.conclusion == 'failure' )
+      name: Zip Trace2 Logs (Windows)
+      run: Compress-Archive -DestinationPath ${{ env.TRACE2_BASENAME }}.zip -Path ${{ env.TRACE2_BASENAME }}
+
+    - name: Archive Trace2 Logs
+      if: ( success() || failure() ) && ( steps.trace2_zip_unix.conclusion == 'success' || steps.trace2_zip_windows.conclusion == 'success' )
```

Comment:
Piggy-backing on @derrickstolee's comment: should this really be also in `success()`' case? If you want to compare between successful runs, that's totally okay, of course. Otherwise, we may save some space.

Reply:
I was fighting with Actions here (or maybe I inherited this from the Azure version).  The `( success() || failure() )` part looks weird, but I found that if I didn't include one of them in the expression, then there was an implicit `success()` that would be inherited (or something like that).  So this was intended as a "I want the logs without regard to whether all the functional tests passed".